python_seed_env/src/seed_env/core.py

```diff
@@ -143,6 +148,31 @@ def seed_environment(self):
     os.makedirs(self.output_dir, exist_ok=True)
     self.output_dir = os.path.abspath(self.output_dir)
 
+    # Determine the template for pyproject.toml. The explicit CLI argument takes precedence.
+    template_path = self.template_pyproject_toml
+    if not template_path and os.path.isfile("./pyproject.toml"):
+      template_path = os.path.abspath("./pyproject.toml")
+      logging.info(
+        f"Found pyproject.toml in the current directory. Using it as a template: {template_path}"
+      )
+
+    # Pre-flight check: Ensure the output directory root is clean of a pyproject.toml, as we will generate one.
+    final_pyproject_path = os.path.join(self.output_dir, "pyproject.toml")
+    if os.path.isfile(final_pyproject_path):
+      # Check for the specific edge case where the output directory is the project root
+      # and the existing pyproject.toml is the one we are using as a template.
+      if template_path and os.path.samefile(template_path, final_pyproject_path):
+        raise FileExistsError(
+          f"The output directory ('{self.output_dir}') contains a 'pyproject.toml', which was found to be used as a template. "
+          "Running this would overwrite the original template file. Please use a different --output-dir; or move the"
+          "existing pyproject.toml to a different location and use the --template-pyproject-toml flag to specify its new location."
```

Comment:
"remove existing pyproject.toml". It is most likely that existing pyproject.toml is a mistake or legacy artifact. If we say "move" it may be understood like the old pyproject.toml is still needed for something.